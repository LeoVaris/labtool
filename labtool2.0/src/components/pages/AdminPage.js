import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button, Loader, Label, Table, Popup } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { Redirect } from 'react-router'
import { HorizontalScrollable } from '../HorizontalScrollable'
import { formatCourseName } from '../../util/format'
import { getAllCI } from '../../services/courseInstance'
import { resetLoading } from '../../reducers/loadingReducer'
import { getAllUsers, updateOtherUser } from '../../services/user'
import { clearNotifications } from '../../reducers/notificationReducer'
import { usePersistedState } from '../../hooks/persistedState'
import { sortUsersBySysopLastname } from '../../util/sort'

const AdminCourseList = ({ courses }) => {
  if (!Array.isArray(courses)) {
    return <div />
  }

  return (
    <HorizontalScrollable>
      <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="1">Course ID</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
            <Table.HeaderCell colSpan="2"> </Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <Table.Body>
          {courses.map(instance => (
            <Table.Row key={instance.id}>
              <Table.Cell>{instance.shorterId} </Table.Cell>
              <Table.Cell>
                <strong>{formatCourseName(instance.name, instance.ohid, instance.start)}</strong>
              </Table.Cell>

              <Table.Cell> {instance.europeanStart} </Table.Cell>
              <Table.Cell textAlign="center">
                <Popup
                  trigger={
                    <Button
                      circular
                      size="tiny"
                      icon={{ name: 'edit', size: 'large', color: 'orange' }}
                      as={Link}
                      to={{
                        pathname: `/labtool/ModifyCourseInstanceStaff/${instance.ohid}`,
                        state: { fromAdmin: true }
                      }}
                    />
                  }
                  position="top right"
                  content="Manage instructors for this course"
                />
              </Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    </HorizontalScrollable>
  )
}

AdminCourseList.propTypes = {
  courses: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
}

const AdminUserList = ({ me, users, updateOtherUser }) => {
  const setAdmin = (userId, name, newAdmin) => () => {
    const message = `Are you sure you want to ${newAdmin ? 'grant admin rights to' : 'remove admin rights from'} ${name}?`
    if (window.confirm(message)) {
      const selfInflicted = userId === me.user.id
      if (selfInflicted && !newAdmin && !window.confirm('You probably do NOT want to remove admin rights from yourself. Are you really sure?')) {
        return
      }
      updateOtherUser({ id: userId, sysop: newAdmin })
    }
  }
  const sortedUsers = sortUsersBySysopLastname(users)

  return (
    <Table singleLine color="yellow" fixed>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Name</Table.HeaderCell>
          <Table.HeaderCell>Status</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {sortedUsers.map(user => (
          <Table.Row key={user.id}>
            <Table.Cell>
              {user.firsts} {user.lastname}
              <br />
              <a href={`mailto:${user.email}`}>{user.email}</a>
            </Table.Cell>
            <Table.Cell>
              {user.teacher && (
                <Label color="orange" horizontal>
                  Teacher
                </Label>
              )}
              {user.sysop ? (
                <div>
                  <Label color="orange" horizontal>
                    Admin
                  </Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, false)} size="tiny" color="green">
                    Remove admin
                  </Button>
                </div>
              ) : (
                <div>
                  <Label horizontal>Not admin</Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, true)} size="tiny" color="green">
                    Make admin
                  </Button>
                </div>
              )}
            </Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  )
}

AdminUserList.propTypes = {
  me: PropTypes.object.isRequired,
  users: PropTypes.array.isRequired,
  updateOtherUser: PropTypes.func.isRequired
}

export const AdminPage = props => {
  const state = usePersistedState('AdminPage', {
    coursesList: false,
    usersList: false
  })

  useEffect(() => {
    props.clearNotifications()
  }, [])

  useEffect(() => {
    if (state.coursesList) {
      props.resetLoading()
      props.getAllCI()
    }
  }, [state.coursesList])

  useEffect(() => {
    if (state.usersList) {
      props.resetLoading()
      props.getAllUsers()
    }
  }, [state.usersList])

  if (!props.user.user.sysop) {
    return <Redirect to={`/labtool`} />
  } else if (props.loading.loading) {
    return <Loader active />
  }

  return (
    <div className="AdminPage">
      <h2>Welcome to the admin interface.</h2>
      <Button compact className={`ui button`} onClick={() => (state.coursesList = !state.coursesList)}>
        {state.coursesList ? 'Hide courses' : 'Manage courses'}
      </Button>
      <br />
      <br />
      {state.coursesList && <AdminCourseList courses={props.courseInstance} />}
      <br />
      <hr />
      <br />
      <Button compact className={`ui button`} onClick={() => (state.usersList = !state.usersList)}>
        {state.usersList ? 'Hide users' : 'Manage users'}
      </Button>
      <br />
      {state.usersList && <AdminUserList me={props.user} users={props.users} updateOtherUser={props.updateOtherUser} />}
    </div>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user,
    courseInstance: state.courseInstance,
    users: state.users,
    loading: state.loading
  }
}

const mapDispatchToProps = {
  getAllCI,
  resetLoading,
  getAllUsers,
  updateOtherUser,
  clearNotifications
}

AdminPage.propTypes = {
  user: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  users: PropTypes.array.isRequired,
  loading: PropTypes.object.isRequired,

  getAllCI: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  updateOtherUser: PropTypes.func.isRequired,
  clearNotifications: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage)
