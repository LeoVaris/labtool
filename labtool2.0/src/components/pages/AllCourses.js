import React, { Component } from 'react'
import { getAll } from '../../services/courseInstance'
import { Link } from 'react-router-dom'

class AllCourses extends Component {
  constructor(props) {
    super(props)
    this.state = {
      courses: [],
    }
  }

  componentDidMount() {
    getAll().then(courses =>
      this.setState({ courses })
    )
  }

  render() {
    return (
      <div>
        <ul>
          {
            this.state.courses.map(c => (
              <li key={c.id}>
                <Link to={`/courses/${c.id}`}>{c.name}</Link>
              </li>
            ))
          }
        </ul>
      </div>
    )
  }
}



export default AllCourses