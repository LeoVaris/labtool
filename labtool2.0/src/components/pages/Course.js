import React, { Component } from 'react'
import { getOne } from '../../services/courseInstance'



class Course extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      course: undefined,
    }
  }

  componentDidMount() {
    getOne(parseInt(this.props.match.params.number, 10)).then(
      course => this.setState({ course })
    )
  }

  render() {
    if (!this.state.course) {
      return <div>course not found!!!</div>
    }

    return (
      <div>
        <h1>{this.state.course.name} </h1>
      </div>
    )
  }
}

export default Course