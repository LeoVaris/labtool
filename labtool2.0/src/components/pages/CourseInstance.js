import React from 'react'

const CourseInstance = ({ instance, handleFieldChange }) => {
  return(
    <li style={{ textAlignVertical: 'center', textAlign: 'center' }}> {instance.name} <button name={instance.name} onClick={handleFieldChange} value={instance.id}>Register</button>
    </li>
  )
}

export default CourseInstance