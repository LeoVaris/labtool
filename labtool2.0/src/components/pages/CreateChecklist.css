.deleteButton {
  float: right;
}

.addForm {
  margin: 0px 10px 10px 10px;
}

.numberField {
  width: 100px;
}

.textField {
  width: 100%;
}

.formField {
  margin-top: 10px;
}

.topOptions {
  margin-bottom: 15px;
}

.saveButtonText {
  font-size: 24px;
}

.deleteButtonText {
  font-size: 14px;
}

.copyForm {
  float: right;
}

.maxPointsCard {
  font-size: 20px;
}

.minimumRequirement .radio label {
  padding-left: 1.35em !important;
}
