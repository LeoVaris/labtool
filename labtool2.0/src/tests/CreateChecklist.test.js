import React from 'react'
import { CreateChecklist } from '../components/pages/CreateChecklist'
import { shallow } from 'enzyme'

describe('<CreateChecklist /> component', () => {
  let wrapper

  const coursePage = {
    id: 10011,
    name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
    start: '2018-03-11T21:00:00.000Z',
    end: '2018-04-29T21:00:00.000Z',
    active: true,
    weekAmount: 7,
    weekMaxPoints: 3,
    currentWeek: 1,
    ohid: 'TKT20010.2018.K.A.1',
    teacherInstances: [
      {
        id: 10001,
        admin: true,
        userId: 10010,
        courseInstanceId: 10011,
        firsts: 'Pää',
        lastname: 'Opettaja'
      },
      {
        id: 10011,
        admin: true,
        userId: 10015,
        courseInstanceId: 10011,
        firsts: 'Ossi Ohjaaja',
        lastname: 'Mutikainen'
      }
    ]
  }
  const checklist = {
    data: {
      Koodi: [
        {
          name: 'Koodin laatu',
          checkedPoints: 2,
          uncheckedPoints: 0,
          textWhenOn: 'Koodin laatu kiitettävää',
          textWhenOff: 'Koodin laadussa parantamisen varaa'
        }
      ],
      Repo: [
        {
          name: 'Readme kunnossa',
          checkedPoints: 0,
          uncheckedPoints: -1,
          textWhenOn: '',
          textWhenOff: 'Readmessa feelua'
        }
      ]
    }
  }

  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <CreateChecklist
        courseId={coursePage.ohid}
        selectedInstance={coursePage}
        checklist={checklist}
        showNotification={mockFn}
        createChecklist={mockFn}
        getOneCI={mockFn}
        getOneChecklist={mockFn}
        resetChecklist={mockFn}
        changeField={mockFn}
        addTopic={mockFn}
        addRow={mockFn}
        removeTopic={mockFn}
        removeRow={mockFn}
      />
    )
  })

  it('renders without error', () => {
    expect(wrapper.find('.CreateChecklist').exists()).toEqual(true)
  })

  describe('Editing form', () => {
    it('autofills text areas', () => {
      const textInputs = wrapper.find('.textField')
      Object.keys(checklist.data).forEach(key => {
        checklist.data[key].forEach(row => {
          let textWhenOnMatch = undefined
          let textWhenOffMatch = undefined
          textInputs.forEach(ti => {
            if (ti.prop('value') === row.textWhenOn) {
              textWhenOnMatch = ti
            }
            if (ti.prop('value') === row.textWhenOff) {
              textWhenOffMatch = ti
            }
          })
          expect(textWhenOnMatch).not.toBe(undefined)
          expect(textWhenOffMatch).not.toBe(undefined)
        })
      })
    })

    it('autofills point values', () => {
      const pointInputs = wrapper.find('.numberField')
      Object.keys(checklist.data).forEach(key => {
        checklist.data[key].forEach(row => {
          let checkedPointsMatch = undefined
          pointInputs.forEach(ti => {
            if (Number(ti.prop('value')) === row.checkedPoints) {
              checkedPointsMatch = ti
            }
          })
          expect(checkedPointsMatch).not.toBe(undefined)
        })
      })
    })
  })

  // It turns out there's no way of testing form functionality in shallow rendering.
  // https://github.com/airbnb/enzyme/issues/308
})
