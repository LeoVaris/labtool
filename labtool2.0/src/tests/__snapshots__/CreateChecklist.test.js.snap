// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateChecklist /> component should render correctly 1`] = `
<Fragment>
  <DocumentTitle
    title="Checklists - Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit"
  />
  <div
    className="CreateChecklist"
  >
    <Connect(BackButton)
      cleanup={[Function]}
      preset="modifyCIPage"
    />
    <Header>
      Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit
    </Header>
    <div
      className="editForm"
    >
      <div
        className="topOptions"
      >
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={true}
          closeOnEscape={true}
          deburr={false}
          icon="dropdown"
          id="weekDropdown"
          minCharacters={1}
          noResultsMessage="No results found."
          onChange={[Function]}
          openOnFocus={true}
          options={Array []}
          placeholder="Select Checklist"
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          selection={true}
          value="week6"
          wrapSelection={true}
        />
        <div
          className="copyForm"
        >
          <Button
            as="button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Copy checklist
          </Button>
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            className="weekDropdown"
            closeOnBlur={true}
            closeOnEscape={true}
            deburr={false}
            disabled={false}
            icon="dropdown"
            minCharacters={1}
            noResultsMessage="No results found."
            onChange={[Function]}
            openOnFocus={true}
            options={Array []}
            placeholder="...from another week"
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            selection={true}
            wrapSelection={true}
          />
           
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            className="courseDropdown"
            closeOnBlur={true}
            closeOnEscape={true}
            deburr={false}
            disabled={false}
            icon="dropdown"
            minCharacters={1}
            noResultsMessage="No results found."
            onChange={[Function]}
            openOnFocus={true}
            options={
              Array [
                Object {
                  "text": "Ohjelmistotekniikan menetelmät (11.03.2018)",
                  "value": 10012,
                },
              ]
            }
            placeholder="...from another course"
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            selection={true}
            value={null}
            wrapSelection={true}
          />
        </div>
        <div
          className="jsonButtons"
        >
          <JsonEdit
            downloadName="TKT20010.2018.K.A.1_week6.json"
            generation={2}
            initialData={
              Object {
                "Koodi": Array [
                  Object {
                    "checkedPoints": 2,
                    "name": "Koodin laatu",
                    "textWhenOff": "Koodin laadussa parantamisen varaa",
                    "textWhenOn": "Koodin laatu kiitettävää",
                    "uncheckedPoints": 0,
                  },
                ],
                "Repo": Array [
                  Object {
                    "checkedPoints": 0,
                    "name": "Readme kunnossa",
                    "textWhenOff": "Readmessa feelua",
                    "textWhenOn": "",
                    "uncheckedPoints": -1,
                  },
                ],
              }
            }
            onImport={[Function]}
          />
        </div>
      </div>
      <div>
        <form
          onSubmit={[Function]}
        >
          <Button
            as="button"
            className="saveButton"
            color="green"
            disabled={true}
            size="large"
            type="submit"
          >
            <div
              className="saveButtonText"
            >
              Save
            </div>
          </Button>
          <br />
          <br />
        </form>
        <div>
          <Card
            color="red"
            fluid={true}
            key="Koodi"
            style={null}
          >
            <CardContent>
              <Header
                color="brown"
                size="huge"
              >
                Koodi
                 
                <Button
                  as="button"
                  className="deleteButton"
                  color="red"
                  onClick={[Function]}
                  size="tiny"
                  type="button"
                >
                  <div
                    className="deleteButtonText"
                  >
                    Delete topic
                  </div>
                </Button>
                <Button
                  as="button"
                  circular={true}
                  className="moveButton"
                  color="blue"
                  disabled={false}
                  icon="arrow circle down"
                  onClick={[Function]}
                  type="button"
                />
                <Button
                  as="button"
                  circular={true}
                  className="moveButton"
                  color="blue"
                  disabled={true}
                  icon="arrow circle up"
                  onClick={[Function]}
                  type="button"
                />
              </Header>
              <div>
                <p>
                  Total points for this section:
                   
                  <strong
                    className="bestPointsNumber"
                  >
                    <Points
                      points={2}
                    />
                  </strong>
                  <span />
                </p>
              </div>
              <div
                className="prerequisiteCategory"
                style={
                  Object {
                    "marginTop": "1em",
                  }
                }
              >
                Set a prerequisite for all checks in this category:
                 
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  className="prerequisiteCategoryDropdown"
                  closeOnBlur={true}
                  closeOnEscape={true}
                  deburr={false}
                  icon="dropdown"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "key": null,
                        "text": "(none)",
                        "value": null,
                      },
                      Object {
                        "key": undefined,
                        "text": "Koodin laatu",
                        "value": undefined,
                      },
                    ]
                  }
                  placeholder="Select prerequisite..."
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  selection={true}
                  wrapSelection={true}
                />
              </div>
            </CardContent>
            <CardContent
              key="Koodin laatu"
            >
              <Header>
                Koodin laatu
                 
                <Button
                  as="button"
                  className="deleteButton"
                  color="red"
                  onClick={[Function]}
                  size="tiny"
                  type="button"
                >
                  <div
                    className="deleteButtonText"
                  >
                    Delete checkbox
                  </div>
                </Button>
                <Button
                  as="button"
                  circular={true}
                  className="moveButtonLarge"
                  color="blue"
                  disabled={true}
                  icon="arrow circle down"
                  onClick={[Function]}
                  type="button"
                />
                <Button
                  as="button"
                  circular={true}
                  className="moveButtonLarge"
                  color="blue"
                  disabled={true}
                  icon="arrow circle up"
                  onClick={[Function]}
                  type="button"
                />
              </Header>
              <div
                className="checkTextRow"
              >
                <div
                  className="checkedLabel"
                >
                  <Label
                    basic={true}
                  >
                    Checked
                  </Label>
                </div>
                <Input
                  className="numberField"
                  label="Points"
                  onBlur={[Function]}
                  onChange={[Function]}
                  step="0.01"
                  type="number"
                  value={2}
                />
                <span
                  style={
                    Object {
                      "flexGrow": 100,
                    }
                  }
                >
                  <Input
                    className="textField"
                    fluid={true}
                    label="Text"
                    onChange={[Function]}
                    type="text"
                    value="Koodin laatu kiitettävää"
                  />
                </span>
              </div>
              <div
                className="checkTextRow"
              >
                <div
                  className="checkedLabel"
                >
                  <Label
                    basic={true}
                  >
                    Unchecked
                  </Label>
                </div>
                <Input
                  className="numberField"
                  label="Points"
                  onBlur={[Function]}
                  onChange={[Function]}
                  step="0.01"
                  type="number"
                  value={0}
                />
                <span
                  style={
                    Object {
                      "flexGrow": 100,
                    }
                  }
                >
                  <Input
                    className="textField"
                    fluid={true}
                    label="Text"
                    onChange={[Function]}
                    type="text"
                    value="Koodin laadussa parantamisen varaa"
                  />
                </span>
              </div>
              <div
                className="minimumRequirement"
                style={
                  Object {
                    "marginTop": "1em",
                  }
                }
              >
                <Checkbox
                  label="Minimum requirement"
                  onChange={[Function]}
                  type="checkbox"
                />
                , which should be
                 
                <Radio
                  checked={false}
                  disabled={true}
                  label="checked"
                  name="minimumRequirementMetIf_Koodi_undefined"
                  onChange={[Function]}
                  type="radio"
                  value={true}
                />
                 
                <Radio
                  checked={false}
                  disabled={true}
                  label="unchecked"
                  name="minimumRequirementMetIf_Koodi_undefined"
                  onChange={[Function]}
                  type="radio"
                  value={false}
                />
                , or else grade will drop by
                 
                <Input
                  className="minimumRequirementGradePenalty"
                  disabled={true}
                  max="5"
                  min="0"
                  onChange={[Function]}
                  step="1"
                  style={
                    Object {
                      "width": "60px",
                    }
                  }
                  type="number"
                />
              </div>
              <div
                className="prerequisite"
                style={
                  Object {
                    "marginTop": "1em",
                  }
                }
              >
                <Label>
                  Prerequisite
                </Label>
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  className="prerequisiteDropdown"
                  closeOnBlur={true}
                  closeOnEscape={true}
                  deburr={false}
                  icon="dropdown"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "key": null,
                        "text": "(none)",
                        "value": null,
                      },
                    ]
                  }
                  placeholder="(none)"
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  selection={true}
                  value={null}
                  wrapSelection={true}
                />
                <Popup
                  content="If a prerequisite is set, the prerequisite must be checked in order for this checkbox to be checkable."
                  disabled={false}
                  eventsEnabled={true}
                  offset={0}
                  on={
                    Array [
                      "click",
                      "hover",
                    ]
                  }
                  pinned={false}
                  position="top left"
                  trigger={
                    <Icon
                      as="i"
                      name="help circle"
                    />
                  }
                />
              </div>
            </CardContent>
            <CardContent>
              <form
                className="addForm"
                onSubmit={[Function]}
              >
                <Button
                  as="button"
                  color="yellow"
                  content="Add new checkbox"
                  icon="plus"
                  labelPosition="left"
                  type="submit"
                />
                <div />
              </form>
            </CardContent>
          </Card>
          <Card
            color="red"
            fluid={true}
            key="Repo"
            style={
              Object {
                "backgroundColor": "#F2F2F2",
              }
            }
          >
            <CardContent>
              <Header
                color="brown"
                size="huge"
              >
                Repo
                 
                <Button
                  as="button"
                  className="deleteButton"
                  color="red"
                  onClick={[Function]}
                  size="tiny"
                  type="button"
                >
                  <div
                    className="deleteButtonText"
                  >
                    Delete topic
                  </div>
                </Button>
                <Button
                  as="button"
                  circular={true}
                  className="moveButton"
                  color="blue"
                  disabled={true}
                  icon="arrow circle down"
                  onClick={[Function]}
                  type="button"
                />
                <Button
                  as="button"
                  circular={true}
                  className="moveButton"
                  color="blue"
                  disabled={false}
                  icon="arrow circle up"
                  onClick={[Function]}
                  type="button"
                />
              </Header>
              <div>
                <p>
                  Total points for this section:
                   
                  <strong
                    className="bestPointsNumber"
                  >
                    <Points
                      points={0}
                    />
                  </strong>
                  <span />
                </p>
              </div>
              <div
                className="prerequisiteCategory"
                style={
                  Object {
                    "marginTop": "1em",
                  }
                }
              >
                Set a prerequisite for all checks in this category:
                 
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  className="prerequisiteCategoryDropdown"
                  closeOnBlur={true}
                  closeOnEscape={true}
                  deburr={false}
                  icon="dropdown"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "key": null,
                        "text": "(none)",
                        "value": null,
                      },
                      Object {
                        "key": undefined,
                        "text": "Readme kunnossa",
                        "value": undefined,
                      },
                    ]
                  }
                  placeholder="Select prerequisite..."
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  selection={true}
                  wrapSelection={true}
                />
              </div>
            </CardContent>
            <CardContent
              key="Readme kunnossa"
            >
              <Header>
                Readme kunnossa
                 
                <Button
                  as="button"
                  className="deleteButton"
                  color="red"
                  onClick={[Function]}
                  size="tiny"
                  type="button"
                >
                  <div
                    className="deleteButtonText"
                  >
                    Delete checkbox
                  </div>
                </Button>
                <Button
                  as="button"
                  circular={true}
                  className="moveButtonLarge"
                  color="blue"
                  disabled={true}
                  icon="arrow circle down"
                  onClick={[Function]}
                  type="button"
                />
                <Button
                  as="button"
                  circular={true}
                  className="moveButtonLarge"
                  color="blue"
                  disabled={true}
                  icon="arrow circle up"
                  onClick={[Function]}
                  type="button"
                />
              </Header>
              <div
                className="checkTextRow"
              >
                <div
                  className="checkedLabel"
                >
                  <Label
                    basic={true}
                  >
                    Checked
                  </Label>
                </div>
                <Input
                  className="numberField"
                  label="Points"
                  onBlur={[Function]}
                  onChange={[Function]}
                  step="0.01"
                  type="number"
                  value={0}
                />
                <span
                  style={
                    Object {
                      "flexGrow": 100,
                    }
                  }
                >
                  <Input
                    className="textField"
                    fluid={true}
                    label="Text"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </span>
              </div>
              <div
                className="checkTextRow"
              >
                <div
                  className="checkedLabel"
                >
                  <Label
                    basic={true}
                  >
                    Unchecked
                  </Label>
                </div>
                <Input
                  className="numberField"
                  label="Points"
                  onBlur={[Function]}
                  onChange={[Function]}
                  step="0.01"
                  type="number"
                  value={-1}
                />
                <span
                  style={
                    Object {
                      "flexGrow": 100,
                    }
                  }
                >
                  <Input
                    className="textField"
                    fluid={true}
                    label="Text"
                    onChange={[Function]}
                    type="text"
                    value="Readmessa feelua"
                  />
                </span>
              </div>
              <div
                className="minimumRequirement"
                style={
                  Object {
                    "marginTop": "1em",
                  }
                }
              >
                <Checkbox
                  label="Minimum requirement"
                  onChange={[Function]}
                  type="checkbox"
                />
                , which should be
                 
                <Radio
                  checked={false}
                  disabled={true}
                  label="checked"
                  name="minimumRequirementMetIf_Repo_undefined"
                  onChange={[Function]}
                  type="radio"
                  value={true}
                />
                 
                <Radio
                  checked={false}
                  disabled={true}
                  label="unchecked"
                  name="minimumRequirementMetIf_Repo_undefined"
                  onChange={[Function]}
                  type="radio"
                  value={false}
                />
                , or else grade will drop by
                 
                <Input
                  className="minimumRequirementGradePenalty"
                  disabled={true}
                  max="5"
                  min="0"
                  onChange={[Function]}
                  step="1"
                  style={
                    Object {
                      "width": "60px",
                    }
                  }
                  type="number"
                />
              </div>
              <div
                className="prerequisite"
                style={
                  Object {
                    "marginTop": "1em",
                  }
                }
              >
                <Label>
                  Prerequisite
                </Label>
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  className="prerequisiteDropdown"
                  closeOnBlur={true}
                  closeOnEscape={true}
                  deburr={false}
                  icon="dropdown"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "key": null,
                        "text": "(none)",
                        "value": null,
                      },
                    ]
                  }
                  placeholder="(none)"
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  selection={true}
                  value={null}
                  wrapSelection={true}
                />
                <Popup
                  content="If a prerequisite is set, the prerequisite must be checked in order for this checkbox to be checkable."
                  disabled={false}
                  eventsEnabled={true}
                  offset={0}
                  on={
                    Array [
                      "click",
                      "hover",
                    ]
                  }
                  pinned={false}
                  position="top left"
                  trigger={
                    <Icon
                      as="i"
                      name="help circle"
                    />
                  }
                />
              </div>
            </CardContent>
            <CardContent>
              <form
                className="addForm"
                onSubmit={[Function]}
              >
                <Button
                  as="button"
                  color="yellow"
                  content="Add new checkbox"
                  icon="plus"
                  labelPosition="left"
                  type="submit"
                />
                <div />
              </form>
            </CardContent>
          </Card>
          <form
            className="addForm"
            onSubmit={[Function]}
          >
            <Button
              as="button"
              content="Add new topic"
              icon="add"
              labelPosition="left"
              size="large"
              type="submit"
            />
            <div />
          </form>
        </div>
        <Card
          className="maxPointsCard"
        >
          <CardContent>
            <p>
              Total points of the checklist:
               
              <strong
                className="totalPointsOfChecklist"
              >
                <Points
                  points={2}
                />
              </strong>
              <span>
                 
                <Popup
                  className="maxPointsIcon"
                  content="The total points don't match the maximum points for this week."
                  disabled={false}
                  eventsEnabled={true}
                  offset={0}
                  on={
                    Array [
                      "click",
                      "hover",
                    ]
                  }
                  pinned={false}
                  position="top left"
                  trigger={
                    <Icon
                      as="i"
                      color="red"
                      name="delete"
                      size="large"
                    />
                  }
                />
              </span>
            </p>
          </CardContent>
          <CardContent>
            <p
              className="maxPointsForWeek"
            >
              Maximum points for this review:
               
              <strong>
                <Points
                  points={3}
                />
              </strong>
            </p>
          </CardContent>
        </Card>
        <div>
          <Label>
            Define maximum points yourself
          </Label>
          <Input
            className="maxPointsInput"
            onChange={[Function]}
            step="0.01"
            style={
              Object {
                "width": "100px",
              }
            }
            type="number"
            value=""
          />
          <Popup
            className="infoText"
            content="The points you define here will be the maximum points for this review. If no value is given, the maximum points for this review 
                    will stay the same as the defaulted weekly points which is 3"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Icon
                as="i"
                name="question circle"
              />
            }
          />
        </div>
        <form
          onSubmit={[Function]}
        >
          <Button
            as="button"
            className="saveButton"
            color="green"
            disabled={true}
            size="large"
            type="submit"
          >
            <div
              className="saveButtonText"
            >
              Save
            </div>
          </Button>
          <br />
          <br />
        </form>
      </div>
    </div>
  </div>
</Fragment>
`;
